tests/test_conversion.py

```diff
@@ -679,3 +679,20 @@ def test_iconvertible_conversion():
     assert 1024 == change_type(1024, System.Int32)
     assert 1024 == change_type(1024, System.Int64)
     assert 1024 == change_type(1024, System.Int16)
+
+def test_intptr_construction():
+    from System import IntPtr, UIntPtr, Int64, UInt64
+
+    ob = ConversionTest()
+
+    assert ob.IntPtrField == IntPtr.Zero
+    assert ob.UIntPtrField == UIntPtr.Zero
+
+    ob.IntPtrField = IntPtr(Int64(-1))
+    assert ob.IntPtrField.ToInt64() == -1
+
+    ob.IntPtrField = IntPtr(Int64(1024))
+    assert ob.IntPtrField.ToInt64() == 1024
+
+    ob.UIntPtrField = UIntPtr(UInt64(1024))
+    assert ob.UIntPtrField.ToUInt64() == 1024
```

Comment:
It should also be possible to construct `UIntPtr` and `IntPtr` from Python `int` type with overflow exception.